custom/enikshay/nikshay_datamigration/management/commands/create_enikshay_cases.py

```diff
@@ -0,0 +1,283 @@
+            episode_structure.case_id = episode_case.case_id
+
+    def create_test_cases(self):
+        if self._outcome:
+            # how many followup's do not have a corresponding outcome? how should we handle this situation?
+            self.factory.create_or_update_cases([self.test(followup) for followup in self._followups])
+
+    @memoized
+    def person(self):
+        return CaseStructure(
+            attrs={
+                'create': True,
+                'case_type': 'person',
+                # 'owner_id': self._location.location_id,
+                'update': {
+                    'aadhaar_number': self.patient_detail.paadharno,
+                    'age_entered': self.patient_detail.page,
+                    'contact_phone_number': validate_number(self.patient_detail.pmob),
+                    'current_address': self.patient_detail.paddress,
+                    'current_address_district_choice': self.patient_detail.Dtocode,
+                    'current_address_state_choice': self.patient_detail.scode,
+                    'first_name': self.patient_detail.first_name,
+                    'last_name': self.patient_detail.last_name,
+                    'middle_name': self.patient_detail.middle_name,
+                    'name': self.patient_detail.pname,
+                    'nikshay_id': self.patient_detail.PregId,
+                    'permanent_address_district_choice': self.patient_detail.Dtocode,
+                    'permanent_address_state_choice': self.patient_detail.scode,
+                    'phi': self.patient_detail.PHI,
+                    'secondary_contact_name_address': (
+                        (self.patient_detail.cname or '')
+                        + ', '
+                        + (self.patient_detail.caddress or '')
+                    ),
+                    'secondary_contact_phone_number': validate_number(self.patient_detail.cmob),
+                    'sex': self.patient_detail.sex,
+                    'tu_choice': self.patient_detail.Tbunitcode,
+
+                    'migration_created_case': True,
+                },
+            },
+        )
+
+    @memoized
+    def occurrence(self, outcome):
+        kwargs = {
+            'attrs': {
+                'create': True,
+                'case_type': 'occurrence',
+                'update': {
+                    'hiv_status': outcome.HIVStatus,
+                    'nikshay_id': outcome.PatientId.PregId,
+
+                    'migration_created_case': True,
+                },
+            },
+            'indices': [CaseIndex(
+                self.person(),
+                identifier='host',
+                relationship=CASE_INDEX_EXTENSION,
+                related_type=self.person().attrs['case_type'],
+            )],
+        }
+
+        for occurrence_case in self.case_accessor.get_cases([
+            index.referenced_id for index in
+            self.case_accessor.get_case(self.person().case_id).reverse_indices
```

Comment:
> It would be better if this assumption wasn't necessary, but the structure of the code is simple enough to check that the assumption is obeyed.

what do you think about adding an explicit assertion just for kicks?